sdk/cosmos/sdk/src/test/java/com/azure/data/cosmos/sync/CosmosSyncItemTest.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ *
+ */
+
+package com.azure.data.cosmos.sync;
+
+import com.azure.data.cosmos.BridgeInternal;
+import com.azure.data.cosmos.CosmosClientBuilder;
+import com.azure.data.cosmos.CosmosContainer;
+import com.azure.data.cosmos.CosmosDatabaseForTest;
+import com.azure.data.cosmos.CosmosItemProperties;
+import com.azure.data.cosmos.CosmosItemRequestOptions;
+import com.azure.data.cosmos.FeedOptions;
+import com.azure.data.cosmos.FeedResponse;
+import com.azure.data.cosmos.PartitionKey;
+import com.azure.data.cosmos.SqlQuerySpec;
+import com.azure.data.cosmos.rx.TestSuiteBase;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.testng.Assert.*;
+
+public class CosmosSyncItemTest extends TestSuiteBase {
+
+    private String preExistingDatabaseId = CosmosDatabaseForTest.generateId();
+    private List<String> databases = new ArrayList<>();
+    private CosmosSyncClient client;
+    private CosmosSyncContainer container;
+
+    @Factory(dataProvider = "clientBuilders")
+    public CosmosSyncItemTest(CosmosClientBuilder clientBuilder) {
+        super(clientBuilder);
+    }
+
+
+    @BeforeClass(groups = {"simple"}, timeOut = SETUP_TIMEOUT)
+    public void beforeClass() {
+        assertThat(this.client).isNull();
+        this.client = clientBuilder().buildSyncClient();
+        CosmosContainer asyncContainer = getSharedMultiPartitionCosmosContainer(this.client.getAsyncClient());
+        container = client.getDatabase(asyncContainer.getDatabase().id()).getContainer(asyncContainer.id());
+    }
+
+    @AfterClass(groups = {"simple"}, timeOut = SHUTDOWN_TIMEOUT, alwaysRun = true)
+    public void afterClass() {
+        assertThat(this.client).isNotNull();
+        this.client.close();
+    }
+
+    @Test(groups = { "simple" }, timeOut = TIMEOUT)
+    public void createItem() throws Exception {
```

Comment:
please also add some tests for valid failure cases (e.g., document already exists 409)
and validate we are throwing the correct exception.

Reply:
Added tests for Conflicts. Will add more tests eventually